bluesky/run_engine.py

```diff
@@ -420,6 +421,17 @@ def __init__(self, md=None, *, loop=None, preprocessors=None,
         self._subscribe_lossless = self.dispatcher.subscribe
         self._unsubscribe_lossless = self.dispatcher.unsubscribe
 
+    def set_gui_mode(self, val):
+        '''
+        Allow the RunEngine to know if it is running on the console (default) or as part of a GUI, if GUI then we want to
+        suppress the RunInterruptedException that is called when the RunEngine is paused because GUI applications have
+        other visual widgets that indicate next steps where as on the console the user must be explicitly told what those
+        options are.
+        :param val:
+        :return:
+        '''
```

Comment:
The docstring may need some rework to match the style of the rest of the docstrings.

Reply:
ok I feel like a dork (appropriate emoji here), not sure why I didn't think of just wrapping the pause/resume/stop calls in a try except block, can you spell tunnel vision?, anyway that solved it fine so I will nuke this PR, thanks @tacaswell for pointing that out. I am very interested in the queue-server so I will look at that tomorrow